cmd/thanos/config.go

```diff
@@ -129,13 +130,24 @@ func (sc *shipperConfig) registerFlag(cmd *kingpin.CmdClause) *shipperConfig {
 type webConfig struct {
 	externalPrefix   string
 	prefixHeaderName string
+	routePrefix      webRoute
+}
+
+type webRoute struct {
+	prefix string
```

Comment:
do we need separate struct? :thinking:

Reply:
I think we do because the web config is used across components but only a couple fo the use the route prefix, so while it is linked to the webconfig it is optional, and if we embed it then we must register it for all if that makes sense?

Did you have an alternative suggestion?

Reply:
I think this is ok :hugs: